File: 2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.cpp
class Solution {
public:
    int maximumProduct(vector<int>& a, int k) {
        
        priority_queue< int, vector< int> , greater< int> > pq;
        long long ans=1,M=1e9+7;
        for( int x:a) pq.push(x);
        
        while( k--){
            int x=pq.top();pq.pop();
            pq.push(x+1);
        }
        
        while(!pq.empty()){
            ans=( ans*pq.top())%M;
            pq.pop();
        }
        return ans; 
        
    }
};

File: 2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.java
class Solution {
    public int maximumProduct(int[] a, int k) {
        
        PriorityQueue<Integer> pq=new PriorityQueue<Integer>();
        long ans=1L,M=1000000007L;

        for( int x:a) pq.add(x);
        
        while(k-- >0){
            pq.add(pq.poll()+1);
        }
        while(!pq.isEmpty()){
            ans=(ans*pq.poll())%M;
        }
        return (int)ans;
    }
}